t/110-take-task-after-reconnect.t

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env tarantool
+
+local fiber = require('fiber')
+local netbox = require('net.box')
+local os = require('os')
+local queue = require('queue')
+local test = require('tap').test()
+local tnt = require('t.tnt')
+
+
+test:plan(1)
+
+local listen = 'localhost:1918'
+tnt.cfg{ listen = listen }
+
+
+local function test_take_task_after_disconnect(test)
+    test:plan(1)
+    local driver = 'fifottl'
+    local tube = queue.create_tube('test_tube', driver,
+        { if_not_exists = true })
+    rawset(_G, 'queue', require('queue'))
+    tube:grant('guest', { call = true })
+    queue.tube.test_tube:put('test_data')
+
+    local connection = netbox.connect(listen)
+    local fiber_1 = fiber.create(function()
+            connection:call('queue.tube.test_tube:take')
+            connection:call('queue.tube.test_tube:take')
+        end)
+
+    fiber.sleep(0.1)
+    connection:close()
+    fiber.set_joinable(fiber_1, true)
```

Comment:
- You can't make joinable fiber that was created via "create". Only "new".
- I suppose you should call "set_joinable" right after fiber.new - without yields

See https://www.tarantool.io/en/doc/1.10/reference/reference_lua/fiber/#lua-function.fiber_object.set_joinable

And one nit: use `fiber1:set_joinable(true)` instead of `fiber.set_joinable(fiber1, true)`

Reply:
Hi, thank for the review. If I understood correctly, I can make a fiber joinable (that was created via create), but "the fiber could become ‘dead’ before fiber_object:set_joinable() takes effect". So, the fiber_1 will be hung on the second call of connection:call('queue.tube.test_tube:take') and don't be a 'dead'. Am I right?

Reply:
Yes, you are right, but let's look at this part again:

```lua
-- this a "main fiber"
local connection = netbox.connect(listen)
-- main fiber yields
local fiber_1 = fiber.create(function()
            -- fiber_1 yields
            connection:call('queue.tube.test_tube:take')
           -- fiber_1 yields
            connection:call('queue.tube.test_tube:take')
end)

-- main fiber yields
fiber.sleep(0.1)
-- There I don't know does main fiber yields on connection.close (I suppose yes, because it's a "I/O action")
-- but not 100% sure
connection:close()
fiber.set_joinable(fiber_1, true)
```

Hmm... I think it could be rewritten with channels is the following manner:

```lua
local ready_to_disconnect = fiber.channel()
local ready_to_take = fiber.channel()

fiber.create(function()
    connection:call('queue.tube.test_tube:take')
    ready_to_disconnect:put({})
    ready_to_take:get()
end)

ready_to_disconnect:get()
connection:close()
ready_to_take:put()
```
Please, consider it, I think it's much simpler and clear

Reply:
Hi. Thank you for the proposal. I am agree that the channels is more elegant decision, but it don't reproduce a bug because you has closed connection before a second call (connection:call('queue.tube.test_tube:take')).